### PR TITLE
Improve the clustermesh status command

### PR DESCRIPTION
**Pull request by giorio94**
This PR includes a set of improvements for the `cilium clustermesh status` command, to simplify troubleshooting and reduce CI flakiness:

* Always check the readiness of the clustermesh-apiserver deployment.
* Report a node as not ready also when the configuration has not yet been detected (as the secret modification has not propagated yet).
* Report explicit errors about why a node is not ready, based on the additional information introduced in https://github.com/cilium/cilium/pull/26788 (e.g., etcd connection, cluster configuration retrieval, data synchronization).
* Remove duplicate checks and fix inaccuracies in output messages.

Please refer to the individual commit messages for additional information.

Fixes: https://github.com/cilium/cilium-cli/issues/1773

~~I'm currently marking this PR as ready for review as it makes sense that the review occurs in parallel with that of https://github.com/cilium/cilium/pull/26788, so that possible modifications can be synchronized. The commit with the go.mod changes will be dropped before merging.~~

### CODE REVIEW

**Review by rolinh — DISMISSED** [review]
It looks like you forgot to remove the commit aptly named "TO DROP: use forked version of cilium"

**Comment by giorio94** [comment]
> It looks like you forgot to remove the commit aptly named "TO DROP: use forked version of cilium"

Yeah, the commit is still there because otherwise the CLI doesn't compile. This is the reason for the `dont-merge/blocked label` being set, as it depends on https://github.com/cilium/cilium/pull/26788. At the same time, I wanted to mark this one ready for review so that they can be reviewed in parallel, and possible changes can be propagated if necessary.

**Review by asauber — COMMENTED** [review]
We're now checking the health of the Deployment, but it may be the case that the Service is separately not able to receive traffic (e.g. an LB Service has not been assigned an IP). Is the idea here that if nodes have synced then the Service must be reachable?

**Comment by giorio94** [comment]
> We're now checking the health of the Deployment, but it may be the case that the Service is separately not able to receive traffic (e.g. an LB Service has not been assigned an IP). Is the idea here that if nodes have synced then the Service must be reachable?

There's actually a service readiness check, but it is hidden inside the `statusAccessInformation` function (which in turn calls `extractAccessInformation`). The inner function returns an error (thus triggering the retry mechanism) if no service IP is found. This is also the reason for the removal of the explicit service readiness check in https://github.com/cilium/cilium-cli/pull/1834/commits/218fc9a24ec9cf0d3630e909255888ea69a063d5, as it was already tested earlier.

**Review by asauber — APPROVED** [review]

File: clustermesh/clustermesh.go [hunk 1]
```diff
@@ -803,12 +804,14 @@ func (k *K8sClusterMesh) extractAccessInformation(ctx context.Context, client k8
...
 		ExternalWorkloadKey:  externalWorkloadKey,
 		ExternalWorkloadCert: externalWorkloadCert,
+		ServiceType:          string(svc.Spec.Type),
 		ServiceIPs:           []string{},
 		Tunnel:               cm.Data[configNameTunnel],
 	}
 
 	switch {
 	case len(endpoints) > 0:
+		svc.Spec.Type = "Manual"
```
Comment (YutaroHayakawa) [inline clustermesh/clustermesh.go:817]: What's this "Manual"?

**Review by YutaroHayakawa — APPROVED** [review]
LGTM with one question

**Pull request merged**